internal/receiver/databricksreceiver/internal/metadata/generated_metrics.go

```diff
@@ -0,0 +1,175 @@
+const Type config.Type = "databricksreceiver"
+
+// MetricIntf is an interface to generically interact with generated metric.
+type MetricIntf interface {
+	Name() string
+	New() pdata.Metric
+	Init(metric pdata.Metric)
+}
+
+// Intentionally not exposing this so that it is opaque and can change freely.
+type metricImpl struct {
+	name     string
+	initFunc func(pdata.Metric)
+}
+
+// Name returns the metric name.
+func (m *metricImpl) Name() string {
+	return m.name
+}
+
+// New creates a metric object preinitialized.
+func (m *metricImpl) New() pdata.Metric {
+	metric := pdata.NewMetric()
+	m.Init(metric)
+	return metric
+}
+
+// Init initializes the provided metric object.
+func (m *metricImpl) Init(metric pdata.Metric) {
+	m.initFunc(metric)
+}
+
+type metricStruct struct {
+	DatabricksJobsActiveTotal     MetricIntf
+	DatabricksJobsRunDuration     MetricIntf
+	DatabricksJobsScheduleStatus  MetricIntf
+	DatabricksJobsTotal           MetricIntf
+	DatabricksTasksRunDuration    MetricIntf
+	DatabricksTasksScheduleStatus MetricIntf
+}
+
+// Names returns a list of all the metric name strings.
+func (m *metricStruct) Names() []string {
+	return []string{
+		"databricks.jobs.active.total",
+		"databricks.jobs.run.duration",
+		"databricks.jobs.schedule.status",
+		"databricks.jobs.total",
+		"databricks.tasks.run.duration",
+		"databricks.tasks.schedule.status",
+	}
+}
+
+var metricsByName = map[string]MetricIntf{
+	"databricks.jobs.active.total":     Metrics.DatabricksJobsActiveTotal,
+	"databricks.jobs.run.duration":     Metrics.DatabricksJobsRunDuration,
+	"databricks.jobs.schedule.status":  Metrics.DatabricksJobsScheduleStatus,
+	"databricks.jobs.total":            Metrics.DatabricksJobsTotal,
+	"databricks.tasks.run.duration":    Metrics.DatabricksTasksRunDuration,
+	"databricks.tasks.schedule.status": Metrics.DatabricksTasksScheduleStatus,
+}
+
+func (m *metricStruct) ByName(n string) MetricIntf {
+	return metricsByName[n]
+}
+
+// Metrics contains a set of methods for each metric that help with
+// manipulating those metrics.
+var Metrics = &metricStruct{
+	&metricImpl{
+		"databricks.jobs.active.total",
+		func(metric pdata.Metric) {
+			metric.SetName("databricks.jobs.active.total")
+			metric.SetDescription("A snapshot of the number of active jobs taken at each scrape")
+			metric.SetUnit("{tot}")
+			metric.SetDataType(pdata.MetricDataTypeGauge)
+		},
+	},
+	&metricImpl{
+		"databricks.jobs.run.duration",
+		func(metric pdata.Metric) {
+			metric.SetName("databricks.jobs.run.duration")
+			metric.SetDescription("The execution duration in milliseconds per completed job")
+			metric.SetUnit("ms")
+			metric.SetDataType(pdata.MetricDataTypeGauge)
+		},
+	},
+	&metricImpl{
+		"databricks.jobs.schedule.status",
+		func(metric pdata.Metric) {
+			metric.SetName("databricks.jobs.schedule.status")
+			metric.SetDescription("A snapshot of the pause/run status per job taken at each scrape")
+			metric.SetUnit("{status}")
```

Comment:
same `{jobs}` suggestion/question

Reply:
There was another {tot} for databricks.jobs.total -- probably what you were referring to, sorry. Changed to {jobs}.

Reply:
My mistake, I thought it was jobs per a given status.